owner/src/main/java/org/the_chance/honeymart/ui/orderdetails/OrderDetailsScreen.kt

```diff
@@ -0,0 +1,126 @@
+package org.the_chance.honeymart.ui.orderdetails
+
+import android.graphics.drawable.Icon
+import android.widget.TextClock
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Devices
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import org.the_chance.design_system.R
+import org.the_chance.honeymart.ui.composables.ContentVisibility
+import org.the_chance.honeymart.ui.products.ProductsContent
+import org.the_chance.honeymart.ui.products.ProductsUiState
+import org.the_chance.honeymart.ui.products.ProductsViewModel
+import org.the_chance.honeymart.ui.products.composables.ProductCard
+import org.the_chance.honeymart.ui.products.contentScreen
+import org.the_chance.honymart.ui.composables.HoneyFilledIconButton
+import org.the_chance.honymart.ui.composables.Loading
+import org.the_chance.honymart.ui.theme.HoneyMartTheme
+import org.the_chance.honymart.ui.theme.black37
+import org.the_chance.honymart.ui.theme.blackOn60
+import org.the_chance.honymart.ui.theme.dimens
+
+@Composable
+fun OrderDetailsScreen(
+    viewModel: OrderDetailsViewModel = hiltViewModel()
+) {
+    val state by viewModel.state.collectAsState()
+    OrderDetailsContent(state)
+}
+
+
+@Composable
+fun OrderDetailsContent(state: OrderDetailsUiState) {
+
+    Loading(state = state.isLoading)
+    ContentVisibility(state = state.contentScreen()) {
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(
+                    top = MaterialTheme.dimens.space24,
+                    start = MaterialTheme.dimens.space16,
+                    end = MaterialTheme.dimens.space16,
+                    bottom = MaterialTheme.dimens.space24,
+                )
+        ) {
+
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.SpaceBetween
+            ) {
+                Column {
+                    Text(
+                        text = "Order #${state.orderDetails.orderId}",
```

Comment:
Do not make any concatenation in the scope of the screen. Any contact should be done in the view model

Reply:
just a correction  it comes form the uistate not the viewmodel

Reply:
okkkkk, I thought this come already from the endpoints with the Order word